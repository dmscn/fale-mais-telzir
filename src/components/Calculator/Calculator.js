import React, { Component } from 'react'

export default class Calculator extends Component {
  render() {
    return (
      <main>
        <h1>Calculator</h1>
      </main>
    )
  }
}
