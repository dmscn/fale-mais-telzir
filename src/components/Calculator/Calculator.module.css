.Container { 
  padding: 1em;
  text-align: center;
}

.Header {
  margin-bottom: 3em;
}