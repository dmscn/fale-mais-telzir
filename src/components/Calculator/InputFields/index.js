export {default} from './InputFields';
export * from './InputFields';