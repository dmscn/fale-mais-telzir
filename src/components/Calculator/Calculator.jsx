// @flow
import React, { Component } from 'react'
import styles from './Calculator.module.css'
import InputFields from './InputFields';
import Result from './Result';

export type State = {
  result?: any
}

export default class Calculator extends Component<null, State> {
  state = {
    result: undefined,
  }

  calculate = ({ tax, plan, time}) => {
    time = parseInt(time);

    let result = {};

    if(plan.descountMin >= time) {
      result.commonPrice = time * tax.pricePerMinute;
      result.promotionPrice = 0;
      this.setState({result});
    } else {
      result.commonPrice = time * tax.pricePerMinute;
      result.promotionPrice = (time - plan.descountMin) * plan.taxIncrease; 
      this.setState({
        result
      });
    }
  }
  
  render() {
    const { result } = this.state;

    if(!result) {
      return (
        <main className={styles.Container}>
            <header className={styles.Header}>
              <h2>Calcule o custo da sua ligação com nossos planos FaleMais</h2>
            </header>
            <InputFields onSubmit={this.calculate}/>
        </main>
      );
    }

    return (
      <main className={styles.Container}>
        <Result result={result} />
      </main>
    )
  }
}
