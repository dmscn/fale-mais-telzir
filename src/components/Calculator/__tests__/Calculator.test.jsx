import React from 'react'
import Calculator from '../Calculator';

describe('<Calculator />', () => {
  it('should render', () => {
    const wrapper = shallow(<Calculator />);
    expect(wrapper).toMatchSnapshot();
  });
});