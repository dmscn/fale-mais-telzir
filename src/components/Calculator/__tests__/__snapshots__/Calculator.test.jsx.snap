// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calculator />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="Header"
        >
          <h2>
            Calcule o custo da sua ligação com nossos planos FaleMais
          </h2>
        </header>,
        <InputFields
          onSubmit={[Function]}
        />,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            Calcule o custo da sua ligação com nossos planos FaleMais
          </h2>,
          "className": "Header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Calcule o custo da sua ligação com nossos planos FaleMais",
          },
          "ref": null,
          "rendered": "Calcule o custo da sua ligação com nossos planos FaleMais",
          "type": "h2",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="Header"
          >
            <h2>
              Calcule o custo da sua ligação com nossos planos FaleMais
            </h2>
          </header>,
          <InputFields
            onSubmit={[Function]}
          />,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              Calcule o custo da sua ligação com nossos planos FaleMais
            </h2>,
            "className": "Header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Calcule o custo da sua ligação com nossos planos FaleMais",
            },
            "ref": null,
            "rendered": "Calcule o custo da sua ligação com nossos planos FaleMais",
            "type": "h2",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
